src/transformers/trainer.py

```diff
@@ -1983,6 +1983,10 @@ def _inner_training_loop(
                     # if loss is nan or inf simply add the average of previous logged losses
                     tr_loss += tr_loss / (1 + self.state.global_step - self._globalstep_last_logged)
                 else:
+                    if tr_loss.device != tr_loss_step.device:
+                        ValueError(
```

Comment:
I think we have to raise here - otherwise I don't think anything will happen 

```suggestion
                        raise ValueError(
```